### PR TITLE
Add migration for `providers.region_code`

### PR DESCRIPTION
**Pull request by stevehook**
## Context

We want to be able to group Providers by region to make large lists easier to navigate.

## Changes proposed in this pull request

- Just a migration for now to add column for `Provider#region_code`

## Guidance to review

- There is still some debate about how we will actually populate this new attribute. The Find API seems to only have values for about a quarter of providers. Is there another way to group providers that we do have data for?
- I don't think there is any need to backfill this data yet and it would have to be done partly from the Find API via the existing syncing mechanism, part manually (not sure who by).

## Link to Trello card

https://trello.com/c/luWz8uHE/975-currently-available-providers-page

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -161,7 +161,7 @@
     t.string "magic_link_token"
     t.datetime "magic_link_token_sent_at"
     t.boolean "hide_in_reporting", default: false, null: false
-    t.integer "course_from_find_id"
+    t.bigint "course_from_find_id"
```
Comment (tvararu) [inline db/schema.rb]: This shouldn't break anything right?

File: db/schema.rb [hunk 2]
```diff
@@ -233,8 +233,8 @@
   create_table "provider_users_providers", force: :cascade do |t|
     t.bigint "provider_id", null: false
     t.bigint "provider_user_id", null: false
-    t.datetime "created_at", default: -> { "now()" }, null: false
-    t.datetime "updated_at", default: -> { "now()" }, null: false
+    t.datetime "created_at", default: -> { "CURRENT_TIMESTAMP" }, null: false
+    t.datetime "updated_at", default: -> { "CURRENT_TIMESTAMP" }, null: false
```
Comment (tvararu) [inline db/schema.rb]: Interesting. I have no reason to believe this won't work, but peculiar that it came up as part of this?

Comment (stevehook) [reply db/schema.rb]: this keeps flip-flopping as different people update the schema. I believe its a Postgres version thing. It's harmless but very irritating. I think I'm going to try upgrading Postgres. I'm on version 10 which is a bit behind some (though it's ahead of production).

Comment (tvararu) [reply db/schema.rb]: 👍 Don't bother. I figured it must be something like that; it's not worth discussing or upheaving your setup, we can roll with it!

File: db/schema.rb (hunk 1 above)
Comment (duncanjbrown) [reply db/schema.rb]: Consensus is not, though we don't completely understand it. See also https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training/pull/1263#discussion_r374214067

File: db/schema.rb (hunk 2 above)
Comment (stevehook) [reply db/schema.rb]: rollback, checkout `schema.rb`, db:reset and re-migrate seems to 'fix it'

**Pull request merged**